examples/src/charmory_examples/object_detection/xview/train_xview_model.py

```diff
@@ -0,0 +1,342 @@
+        boxes = torch.as_tensor(box_final, dtype=torch.float32)
+        area = img['objects']['area']
+        area = torch.as_tensor(area, dtype=torch.float32)
+        iscrowd = torch.zeros((boxes.shape[0],), dtype=torch.int64)
+        labels = img['objects']['category']
+        labels = list(
+                map(lambda x: LABEL_MAP[x], labels)
+            )  # map original classes to sequential classes
+        labels = torch.as_tensor(labels, dtype=torch.int64)
+
+        target = {}
+        target["boxes"] = boxes
+        target["labels"] = labels
+        target["area"] = area
+        target["iscrowd"] = iscrowd
+        image_id = torch.tensor([idx])
+        target["image_id"] = image_id
+
+
+        if self.transforms:
+            sample = self.transforms(image = image_resized,
+                                     bboxes = target['boxes'],
+                                     labels = labels)
+            image_resized = sample['image']
+            target['boxes'] = torch.Tensor(sample['bboxes'])
+            
+        return image_resized, target
+
+
+
+    def __len__(self):
+        return len(self.image_dataset)
+
+from torch.utils.data import Dataset, DataLoader
+def create_train_loader(train_dataset, num_workers=2):
+    train_loader = DataLoader(
+        train_dataset,
+        batch_size=BATCH_SIZE,
+        shuffle=True,
+        num_workers=num_workers,
+        collate_fn=collate_fn
+    )
+    return train_loader
+def create_valid_loader(valid_dataset, num_workers=2):
+    valid_loader = DataLoader(
+        valid_dataset,
+        batch_size=BATCH_SIZE,
+        shuffle=False,
+        num_workers=num_workers,
+        collate_fn=collate_fn
+    )
+    return valid_loader
+import torchvision
+from torchvision.models.detection.faster_rcnn import FastRCNNPredictor,FasterRCNN_MobileNet_V3_Large_320_FPN_Weights
+
+model = torchvision.models.detection.fasterrcnn_mobilenet_v3_large_320_fpn(weights=FasterRCNN_MobileNet_V3_Large_320_FPN_Weights.DEFAULT)
+
+def create_model(num_classes):
+    num_classes = num_classes   # 63 classes
+    # get number of input features for the classifier
+    in_features = model.roi_heads.box_predictor.cls_score.in_features
+    # replace the pre-trained head with a new one
+    model.roi_heads.box_predictor = FastRCNNPredictor(in_features, num_classes)
+    return model
+
+train_dataset_1 = xview(train_dataset, 3000, 3000,  transforms=get_train_transform())
+test_dataset_1 = xview(test_dataset, 3000, 3000,  transforms=get_valid_transform())
+train_loader = create_train_loader(train_dataset_1)
+valid_loader = create_valid_loader(test_dataset_1)
+print(f"Number of training samples: {len(train_dataset)}")
+print(f"Number of validation samples: {len(test_dataset)}\n")
+# initialize the model and move to the computation device
+model = create_model(6)
+model = model.to(DEVICE)
+
+
+from tqdm.auto import tqdm
+
+# function for running training iterations
+def train(train_data_loader, model):
```

Comment:
Is there a particular reason this is done manually instead of using a Lightning Trainer?

Reply:
I did try to do this, but the lightning file was a pretty similar structure as this file. Meaning I had to create another class that was a child class of the LightningModule that was the only main difference. I don't think I ever got it completely working b/c there were some issues that I couldn't figure out. I was able to get this one working pretty easily. Do you have a clean example that you have from one of your experiments?

Reply:
I did try to do this, but the lightning file was a pretty similar structure as this file. Meaning I had to create another class that was a child class of the LightningModule that was the only main difference. I don't think I ever got it completely working b/c there were some issues that I couldn't figure out. I was able to get this one working pretty easily. Do you have a clean example that you have from one of your experiments? I could try to modify that file to get lighting training working